akka-docs/src/test/java/jdocs/stream/HubDocTest.java

```diff
@@ -137,4 +139,91 @@ public void mergeBroadcastCombination() {
     killSwitch.shutdown();
     //#pub-sub-4
   }
+  
+  @Test
+  public void dynamicPartition() {
+    // Used to be able to clean up the running stream
+    ActorMaterializer materializer = ActorMaterializer.create(system);
+
+    //#partition-hub
+    // A simple producer that publishes a new "message-n" every second
+    Source<String, Cancellable> producer = Source.tick(
+      FiniteDuration.create(1, TimeUnit.SECONDS),
+      FiniteDuration.create(1, TimeUnit.SECONDS),
+      "message"
+    ).zipWith(Source.range(0, 100), (a, b) -> a + "-" + b);
+
+    // Attach a PartitionHub Sink to the producer. This will materialize to a
+    // corresponding Source.
+    // (We need to use toMat and Keep.right since by default the materialized
+    // value to the left is used)
+    RunnableGraph<Source<String, NotUsed>> runnableGraph =
+      producer.toMat(PartitionHub.of(
+          String.class, 
+          (size, elem) -> Math.abs(elem.hashCode()) % size,
+          2, 256), Keep.right());
```

Comment:
Something fishy with indentation here, 2 and 256 are parameters to `of` but then it ends, could we indent the different parameter lists with different levels to make it easier to read?

Something like:
```
    producer.toMat(
      PartitionHub.of(
        String.class, 
        (size, elem) -> Math.abs(elem.hashCode()) % size,
        2, // startAfterNbrOfConsumers
        256), // bufferSize
      Keep.right());
```